### DISCUSSION TITLE
Sumo tlLogic parameter

### DISCUSSION THREAD

**Discussion by Cheng-Yuan-0124** [eclipse/mosaic#220] [Q&A] [answered]
Hello all,

I am doing the job of integrating the CARLA with MOSAIC, one of the problem I met is when I use the CARLA netconverter script to generate SUMO net.xml, it will come with a linkSignalID param in the tlLogic section (example below). The net.xml file works good if I only use SUMO GUI to load and run. However, it is not acceptable for the MOSAIC to read. Is there any reason why MOSAIC not allow this type of parameter in the tlLogic?

  ```
<tlLogic id="148" type="static" programID="0" offset="0">
    <phase state="ggrrrrrrrgrr" duration="42"/>
    <phase state="yyrrrrrrryrr" duration="3"/>
    <phase state="rrrrrrrrrrrr" duration="3"/>
    <phase state="rrggrrrrrrrg" duration="42"/>
    <phase state="rryyrrrrrrry" duration="3"/>
    <phase state="rrrrrrrrrrrr" duration="3"/>
    <phase state="rrrrrrrggrgr" duration="42"/>
    <phase state="rrrrrrryyryr" duration="3"/>
    <phase state="rrrrrrrrrrrr" duration="3"/>
    <phase state="rrrrgggrrrrr" duration="42"/>
    <phase state="rrrryyyrrrrr" duration="3"/>
    <phase state="rrrrrrrrrrrr" duration="3"/>
    <param key="linkSignalID:0" value="1607"/>
    <param key="linkSignalID:1" value="1607"/>
    <param key="linkSignalID:2" value="1609"/>
    <param key="linkSignalID:3" value="1609"/>
    <param key="linkSignalID:4" value="1608"/>
    <param key="linkSignalID:5" value="1608"/>
    <param key="linkSignalID:6" value="1608"/>
    <param key="linkSignalID:7" value="1610"/>
    <param key="linkSignalID:8" value="1610"/>
    <param key="linkSignalID:9" value="1607"/>
    <param key="linkSignalID:10" value="1610"/>
    <param key="linkSignalID:11" value="1609"/>
  </tlLogic>
```

**Comment by schwepmo** [comment, marked as answer]
Hello,
this indeed seems to be an issue with MOSAICs TraCI-coupling. One solution would be to use the libsumo-coupling ([see here](https://www.eclipse.org/mosaic/docs/simulators/traffic_simulator_sumo/#using-libsumo-with-eclipse-mosaic)). 
Alternatively you can try to apply the [patch](https://github.com/eclipse/mosaic/files/8823858/fix_traci_tl_program_reader.zip) included in this zip, which should fix your issue.
We will further address this in the near future. ✌️ 

PS: Internally we already worked on a CARLA coupling, which will probably be included in MOSAIC in the autumn release

**Reply by Cheng-Yuan-0124** [reply]
Thanks for the information, that helps a lot. One more question, could you give me some information about what CARLA version you are planning to use for the integration with MOSAIC and the date you ideally to release?

**Reply by schwepmo** [reply]
Currently we are working with version `0.9.13`.
As our coupling is still very much a work in progress and we haven't decided how much we will release to open-source I sadly cannot provide you with a release date.
However you can try sending a mail to mosaic@fokus.fraunhofer.de and we can maybe provide you with our code depending on your use-case.
